pkg/sql/internal_test.go

```diff
@@ -582,6 +582,47 @@ func TestInternalExecutorInLeafTxnDoesNotPanic(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestInternalExecutorWithDefinedQoSOverrideDoesNotPanic(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	s, _, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer s.Stopper().Stop(ctx)
+
+	ie := s.InternalExecutor().(*sql.InternalExecutor)
+	qosLevel := sessiondatapb.TTLLow
+	_, err := ie.ExecEx(
+		ctx, "defined_quality_of_service_level_does_not_panic", nil,
+		sessiondata.InternalExecutorOverride{User: security.RootUserName(), QualityOfService: &qosLevel},
```

Comment:
is there a reason to pass this in by pointer? the default is 0 which is normal pri.